sale_order_line_cancel/models/sale_order_line.py

```diff
@@ -57,7 +58,15 @@ def _update_qty_canceled(self):
         for line in self:
             if line._get_moves_to_cancel():
                 continue
-            line.product_qty_canceled = line.qty_to_deliver
+            if (
+                line.state == "sale"
+                and line.company_id.on_sale_line_cancel_decrease_line_qty
+            ):
+                line.product_uom_qty = max(
```

Comment:
@mt-software-de IFAIK, if you update the product_uom_qty on a confirmed SO, this will trigger procurement rules with a negative qty for the product....

Reply:
It shouldn't because the outgoing qty is the same AS the product_uom_qty https://github.com/odoo/odoo/blob/16.0/addons%2Fsale_stock%2Fmodels%2Fsale_order_line.py#L329